### ISSUE TITLE
Jetpack Cloud: `SiteSelector` in sidebar cannot be opened by keyboard

### ISSUE THREAD

**Issue by mirka** [Automattic/wp-calypso#102390]
### Context and steps to reproduce

In the Jetpack Cloud sidebar, the site selector at the top cannot be opened by keyboard.

https://github.com/user-attachments/assets/537b804a-e0bf-4e5d-9764-c33e9176a692

In fixing keyboard accessibility, the markup should also be reviewed as it is incorrect for a trigger button like this (currently an href-less anchor).

### Site owner impact

Fewer than 20% of the total website/platform users

### Severity

Critical

**Comment by Robertght** [comment]
Thank you for reporting this @mirka !

I could replicate this on my end as well so I've marked it as Triaged.
